gov-ithub/socent-frontend
# Authentication session handling #1

#### Fixes #54 

#### Here are the changes I propose:
Start by having `API.js` handle authentication. This means we can handle local storage for the token in one place, while also passing the token down to `Entity.js` children so they can make authenticated calls to the backend. Also deny access to components that should only be available to admins. 

There's more work to do to ensure correct session handling for the token, but this is a decent start before it gets out of hand. Notably, sign out is not implemented yet (use dev console to clear local storage).

#### Test plan:
- Navigate to `admin/`
- Validate that you're redirected to `login/`
- Login with the provided username/password
- Confirm you're being redirected to the admin dashboard
- Confirm the API calls contain an authorization token
- Confirm that clearing local storage redirects you to `login/` upon page refresh or route change

#### Suggested reviewers: @gov-ithub/socent, @cezarneaga
